### PR TITLE
implementation of the method getStatistic

### PR DESCRIPTION
**Pull request by BusyhinD**

### CODE REVIEW

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,8 +1,71 @@
...
+            while (bufferedReader.ready()) {
+                stringBuilder.append(bufferedReader.readLine()).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
+        }
+        return stringBuilder.toString();
+    }
+
+    private String reportCreation(String fileData) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:32]:
```suggestion
    private String createReport(String data) {
```
Why 'fileData'? It is not important for this method where data came from.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,8 +1,71 @@
...
+
+    private String readFromFile(String fileName) {
+        StringBuilder stringBuilder;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            stringBuilder = new StringBuilder();
+            while (bufferedReader.ready()) {
+                stringBuilder.append(bufferedReader.readLine()).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:27]:
```suggestion
            throw new RuntimeException("Can't read file: " + fileName, e);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,8 +1,71 @@
...
+                .append(System.lineSeparator())
+                .append(stringResult).append(separator).append((supplyCounter - buyCounter))
+                .toString();
+    }
 
+    private void writeToFile(String fileName, String report) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(fileName))) {
+            bufferedWriter.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to file", e);
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:68]:
```suggestion
            throw new RuntimeException("Can't write to file: " + fileName, e);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,8 +1,71 @@
...
+            if (datumArray[DATA_TYPE_INDEX].contains(stringBuy)) {
+                buyCounter += Integer.parseInt(datumArray[DATA_VALUE_INDEX]);
+            } else {
+                supplyCounter += Integer.parseInt(datumArray[DATA_VALUE_INDEX]);
+            }
+        }
+        return dataBuilding(buyCounter, supplyCounter);
+    }
+
+    private String dataBuilding(int buyCounter, int supplyCounter) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:49]:
```suggestion
    private String buildReport(int bought, int supplied) {
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,8 +1,71 @@
...
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int DATA_TYPE_INDEX = 0;
+    private static final int DATA_VALUE_INDEX = 1;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileData = readFromFile(fromFileName);
+        String report = reportCreation(fileData);
```
Comment (aaananas) [inline src/main/java/core/basesyntax/WorkWithFile.java:15]:
```suggestion
        String report = createReport(fileData);
```
methods names should follow this pattern `doSomething()`

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**